### DISCUSSION TITLE
Allow override for table names

### DISCUSSION THREAD

**Discussion by erikkraijenoord** [codeigniter4/shield#585] [Ideas]
Shield is as far as I know very customizable as a package, but one feature seems to be missing.. With for example IonAuth you can change table names to prevent conflict with other authorization packages. Or you just have different naming policy, however it is not possible to override these names.

Q: Why can we change; views, login method & much more but not the table names?

----
On the [forum thread](https://forum.codeigniter.com/showthread.php?tid=82321) i've given a simple example (i'm superior).
Would this be something you would consider as a feature request?

```php
/**
 * --------------------------------------------------------------------
 * Shield Tables
 * --------------------------------------------------------------------
 *
 */
public array $table = [
    'groups_users'      => 'auth_groups_users',
    'identities'        => 'auth_identities',
    'logins'            => 'auth_logins',
    'permissions_users' => 'auth_permissions_users',
    'remember_tokens'   => 'auth_remember_tokens',
    'token_logins'      => 'auth_token_logins',
];

// Example usage:
config('Auth')->table['groups_users'];

**Comment by datamweb** [comment]
Hi, I remember your request in the forum,
I have also used IonAuth and I know it has this feature.

We are in **beta** phase. And we welcome any feedback to improve Shield. We did not have a high demand in this regard.
However, Yes, the Shield is flexible, in my opinion, this feature can be added to the **Shield**.

+1 from me, wait for other members and users to decide on this.

**Comment by lonnieezell** [comment]
I have no problem with this being added.

**Comment by kenjis** [comment]
Adding the table property is no problem.

If you need to change a table name right now, you could do it.
Extend a model class and change the table property.

**Comment by kenjis** [comment]
See #633
